src/main/java/frc/robot/commands/AutoBalance.java

```diff
@@ -47,39 +47,48 @@ public void initialize() {
   @Override
   public void execute() {
     // Gets how far the robot is tilted and calculates proper drive power from it
-    m_currentAngle = m_drive.getGyroRoll().getDegrees();
+    m_currentAngle = m_drive.getGyroPitch().getDegrees();
     m_error = AutoConstants.DESIRED_BALANCE_ANGLE - m_currentAngle;
 
     m_drivePower = Math.min(m_balanceController.calculate(m_currentAngle), 1);
     // Limit max power
-    m_drivePower = MathUtil.clamp(m_drivePower, -0.2, 0.2);
+    m_drivePower = MathUtil.clamp(m_drivePower, -0.6, 0.6);
+
+    double rollRateBalance = m_drive.getGyroRollPitch();
+    if (Math.abs(rollRateBalance) >= 0.3) {
+      m_drivePower = Math.copySign(0.0, -m_drive.getGyroPitch().getDegrees());
+      SmartDashboard.putNumber("Stop Balancing", 1);
+    }
 
     // Counter for checking if robot is truly balanced
-   if (Math.abs(m_currentAngle) < AutoConstants.DESIRED_BALANCE_ANGLE) {
-     m_timer.start();
-   }
-   if (Math.abs(m_currentAngle) < AutoConstants.DESIRED_BALANCE_ANGLE && m_timer.get() >= 10) {
-     m_timer.stop();
-     m_isLevel = true;
-   }
-   if (m_timer.get() > 0 && Math.abs(m_currentAngle) > AutoConstants.DESIRED_BALANCE_ANGLE) {
-     m_timer.stop();
-     m_timer.reset();
-   }
-
-    m_drive.setModuleStates(m_drivePower, 0.0, 0.0);
+    if (Math.abs(m_currentAngle) < AutoConstants.DESIRED_BALANCE_ANGLE) {
+      m_timer.start();
+      m_drivePower = 0.0;
+    }
+    if (Math.abs(m_currentAngle) < AutoConstants.DESIRED_BALANCE_ANGLE && m_timer.get() >= 10) {
```

Comment:
These if statements should be mutually exclusive. (Meaning they should be else ifs.)